docs/DesignDocs/RemoteAttestationCollaterals.md

```diff
@@ -0,0 +1,292 @@
+ */
+typedef struct _oe_collaterals_t
+{
+    uint32_t version;
+    uint8_t enclave_type;         ///< The type of enclave
+    uint8_t reserved;
+    uint16_t reserved2;
+    uint32_t num_elements;
+    size_t buffer_size;           ///< Size of the buffer
+
+    ///< Data buffer is made of an offset array of type uint32_t, followed by
+    ///< the actual data.
+    ///< This array has the size of **num_elements** and stores the offset
+    ///< into the data section.
+    ///<  _________________________
+    ///<  |  version              |
+    ///<  |-----------------------|
+    ///<  |  reserved    | type   |
+    ///<  |-----------------------|
+    ///<  |  num_elements         |
+    ///<  |-----------------------|
+    ///<  |  buffer_size          |
+    ///<  |-----------------------|
+    ///<  |                       |
+    ///<  |  array(uint32_t)      |
+    ///<  |  size of num_elements |
+    ///<  |-----------------------|
+    ///<  |  Data                 |
+    ///<  |_______________________|
+    ///<
+    uint8_t buffer[];              ///< Buffer of offsets + data
+
+} oe_collaterals_t;
+```
+
+### New Collateral functions
+
+`collaterals.c`
+```C
+/*! \struct sgx_collaterals
+ *
+ * \brief SGX collateral structure
+ *
+ * The generic oe_collaterals_t structure is parsed and converted into this
+ * internal structure.  The order of the generic data elements should
+ * coincide with the order of the fields in this structure.
+ */
+typedef struct _sgx_collaterals_t
+{
+    uint8_t* tcb_info;                      ///< TCB info
+    size_t tcb_info_size;                   ///< TCB Info size
+    uint8_t* tcb_issuer_chain;              ///< PEM format
+    size_t tcb_issuer_chain_size;           ///< Size of the tcb_issuer_chain
+    uint8_t* crl[3];                        ///< CRLs
+    size_t crl_size[3];                     ///< CRLs sizes
+    uint8_t* crl_issuer_chain[3];           ///< PEM format
+    size_t crl_issuer_chain_size[3];        ///< Size of each crl_issuer_chain
+
+    uint8_t* qe_id_info;                    ///< QE Identity info
+    size_t qe_id_info_size;                 ///< QE Identity size
+    uint8_t* qe_id_issuer_chain;            ///< PEM format
+    size_t qe_id_issuer_chain_size;         ///< Size of qe_id_issuer_chain
+
+    uint8_t* creation_datetime;             ///< Time the collaterals were generated
+    size_t create_datetime_size;            ///< The size of creation_datetime.  Should be 24.
+
+} sgx_collaterals_t;
+
+/**
+ * Returns the collaterals needed for verification of a remote OE report.
+ *
+ * This function is only available inside an enclave.
+ *
+ * @param[in] enclave The instance of the enclave that will be used.
+ * @param[in] collaterals_buffer The buffer containing the collaterals to parse.
+ * @param[in] collaterals_buffer_size The size of the **collaterals_buffer**.
+ *
+ * @retval OE_OK The collaterals were successfully retrieved.
+ */
+oe_result_t oe_get_collaterals(
```

Comment:
Would this function be used by the challenger as well for the Scenario 3? If so, wondering if it should have an input parameter where the policy can be specified.

Reply:
Same as above.  I am not familiar with the "user generated policy". Maybe we can handle that as a separate issue/PR?

Reply:
What you are asking can be supported using the upcoming plug-in model.